### PR TITLE
This change fixes issue #26 and #42

### PR DESCRIPTION
**Pull request by canelhasmateus**

### CODE REVIEW

File: src/nimvscode/nimSuggestExec.nim [hunk 1]
```diff
@@ -29,7 +29,7 @@ type NimSuggestProcessDescription* = ref object
 
 var nimSuggestPath: cstring
 var nimSuggestVersion: cstring
-var nimSuggestProcessCache = newJsAssoc[cstring, Future[
+var nimSuggestProcessCache = newMap[cstring, Future[
```
Comment (saem) [inline src/nimvscode/nimSuggestExec.nim:32]: This is better, but I'm wondering if there is an issue on line 157 with the jsDelete.

**Review by saem — CHANGES_REQUESTED** [review]

File: src/nimvscode/nimSuggestExec.nim (hunk 1 above)
Comment (canelhasmateus) [reply src/nimvscode/nimSuggestExec.nim:32]: It removes the key from the process map right?
As far as i understand, this should only be done if the closeCachedProcess is sucessful. is this what you're refering to?

Comment (saem) [reply src/nimvscode/nimSuggestExec.nim:32]: `jsDelete` is equivalent to the `delete` keyword in JS.

There is a `delete` proc on `Map` itself, checkout `jsNode.nim`

Comment (canelhasmateus) [reply src/nimvscode/nimSuggestExec.nim:32]: considering its deleting only the entry ( which was just closed and undefined ) , i don't think it matters.
edit: that is , if it is called after sucessfully closed. as its stands, its still possible to leak processes, if closeCachedProcess raises.

Comment (saem) [reply src/nimvscode/nimSuggestExec.nim:32]: This: https://nim-lang.org/docs/jsffi.html#jsDelete%2Cauto

Which is this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete

Is not this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/delete

Which should be this: https://github.com/saem/vscode-nim/blob/681d54cc3647186c174f173677f80a7e98ffee23/src/nimvscode/jsNode.nim#L80

Comment (saem) [reply src/nimvscode/nimSuggestExec.nim:32]: There are some JS shenanigans going on here, so if you're not familiar that's ok.

The gist is that one is an old school operator and Map which is a newer API, has a method on it.

Comment (canelhasmateus) [reply src/nimvscode/nimSuggestExec.nim:32]: thats fine, i'll be changing the implementation here

File: src/nimvscode/nimSuggestExec.nim [hunk 2]
```diff
@@ -154,7 +154,7 @@ proc closeNimsuggestProcess*(file: cstring) {.async.} =
       console.error("closeCachedProcess failed", getCurrentException())
     finally:
       nimSuggestProcessCache[file] = jsUndefined.to(Promise[NimSuggestProcessDescription])
-      discard jsDelete nimSuggestProcessCache[file]
+      discard delete(nimSuggestProcessCache, file)
```
Comment (saem) [inline src/nimvscode/nimSuggestExec.nim]:
```suggestion
      discard nimSuggestProcessCache.delete(file)
```

There is no return value in my import of it but there should be one with `bool` -- that can be fixed later.

The `discard` is still good to keep for that reason.

**Comment by saem** [comment]
This was some solid detective work, nice!

**Pull request merged**